browser-switch/src/main/java/com/braintreepayments/browserswitch/BrowserSwitchOptions.java

```diff
@@ -0,0 +1,89 @@
+package com.braintreepayments.browserswitch;
+
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.fragment.app.FragmentActivity;
+
+import org.json.JSONObject;
+
+/**
+ * Parameter object that contains a set of BrowserSwitch parameters for use with
+ * {@link BrowserSwitchClient#start(BrowserSwitchOptions, FragmentActivity, BrowserSwitchListener)}
+ * and related convenience methods.
+ */
+public class BrowserSwitchOptions {
+
+    private Intent intent;
+    private JSONObject metadata;
+    private int requestCode;
+    private Uri url;
+
+    /**
+     * Set browser switch intent
+     * @param intent The target intent to use for browser switch. Required unless url specified
+     * @return {@link BrowserSwitchOptions} returns reference to instance to allow setter invocations to be chained
+     */
+    public BrowserSwitchOptions intent(Intent intent) {
+        this.intent = intent;
+        return this;
+    }
+
+    /**
+     * Set browser switch metadata
+     * @param metadata JSONObject containing metadata to associate with the browser switch request
```

Comment:
I wonder if we could expand this doc comment a bit to explain that this metadata property is intended for an information that you want to have available when returning from the browser. We could also state that it will be persisted even if the app is terminated during the browser switch.

```suggestion
     * @param metadata JSONObject containing metadata necessary for handling the return from browser to app. This data will be persisted and returned in {@link BrowserSwitchResult} even if the app is terminated during the browser switch.
```

Reply:
Yeah I like that. It's good to explain the feature more.